setuptools/__init__.py

```diff
@@ -149,6 +150,7 @@ def finalize_options(self):
 
 def setup(**attrs):
     # Make sure we have any requirements needed to interpret 'attrs'.
+    logging.configure()
```

Comment:
I wonder if this function should also be called from the 'build_meta'... (in theory for PEP517-style buils that is the first piece of setuptools coda that run, right?)

Reply:
Good tip. I looked briefly and wasn't sure where to invoke that behavior. But moreover, I was concerned that adding it in build_meta would cause it to be called twice (once when build_meta is setup and again when setup.py is executed by build_meta). My instinct was that this hook would happen exactly once, and if a future refactor went to remove setup.py, this call could be moved then.  I also note that for distutils, logging didn't get setup until somewhere around Distribution.finalize_options, so this configuration will happen even earlier than for distutils. It might be nice for it to happen even earlier, so it could affect code in setup.py prior to invoking setup(), but I'm okay with this for now.